### ISSUE TITLE
code should use entropy difference, but uses perplexity difference

### ISSUE THREAD

**Issue by rsennrich** [lukeorland/moore_and_lewis_data_selection#2]
Hi Luke,

your algorithm computes the difference in perplexity, whereas Moore and Lewis define their filter as the difference of the cross entropy (i.e. the log probability normalized by sentence length). This will result in a very different ranking - if you want to operate with perplexities, you should calculate log(ppl(x)/ppl(y)) to obtain the cross-entropy difference (H(x)-H(y)). Alternatively, you can take the logprob output from SRILM and divide that by the number of words to obtain the cross-entropy.

**Comment by amirkamran** [comment]
Yes this is correct the difference of perplexity is resulting in poor selection.
